src/main/java/uk/gov/pay/adminusers/resources/UserResource.java

```diff
@@ -79,6 +80,30 @@ public Response getUser(@PathParam("externalId") String externalId) {
                 .orElseGet(() -> Response.status(NOT_FOUND).build());
     }
 
+    @Path(USER_EMAILS_RESOURCE)
+    @POST
+    @Produces(APPLICATION_JSON)
+    @Consumes(APPLICATION_JSON)
+    public Response getUserEmails(JsonNode payload) {
+        if (payload != null && payload.get(GetUserProfilesRequest.FIELD_EXTERNAL_IDS) != null) {
+            logger.info("Get UserProfiles request - [ {} ]", payload.toString());
+
+            List<String> externalIds = newArrayList(payload.get(GetUserProfilesRequest.FIELD_EXTERNAL_IDS).elements())
+                    .stream().map(externalIdNode -> externalIdNode.textValue()).collect(Collectors.toList());
```

Comment:
Rather than creating a temporary `ArrayList` and streaming that, you could use Guava’s `Streams`:

```
Streams.stream(payload.get(GetUserProfilesRequest.FIELD_EXTERNAL_IDS)).map(JsonNode::textValue).collect(Collectors.toList());
```